src/hermit.rs

```diff
@@ -115,4 +143,60 @@ mod tests {
         assert_eq!(*first_op,
                    Op::GitInit(PathBuf::from("/home/geoff/.hermit-config/shells/new-one")));
     }
+
+    #[test]
+    fn can_inhabit_existing_shell() {
+        let config = MockConfig {
+            root_path: PathBuf::from(".config"),
+            allowed_shell_names: vec!["default".to_owned(), 
+                "new-one".to_owned()],
+            current_shell: "default".to_owned(),
+        };
+        let mut hermit = hermit(&config);
+        let mut file_ops = FileOperations::rooted_at("/home/houngj"); 
+        hermit.inhabit_shell(&mut file_ops, "new-one");
+        let shell = hermit.current_shell().unwrap();
+        assert_eq!(shell.name, "new-one"); 
+    }
+    
+    #[test]
+    fn cant_inhabit_nonexisting_shell() {
+        let config = MockConfig {
+            root_path: PathBuf::from(".config"),
+            allowed_shell_names: vec!["default".to_owned()],
+            current_shell: "default".to_owned(),
+        };
+        let mut hermit = hermit(&config);
+        let mut file_ops = FileOperations::rooted_at("/home/houngj");
+        let res = hermit.inhabit_shell(&mut file_ops, "not-a-shell");
+        assert!(res.is_err());
+        assert_eq!(res.err().unwrap(), Error::ShellDoesNotExist);
+    }
+    
+    #[test]
+    fn can_get_inhabitable_shells() {
+        let mut test_root = set_up("inhabit-test");
+        let config = MockConfig{
+            root_path: PathBuf::from(&test_root),
+            allowed_shell_names: vec!["default".to_owned()],
+            current_shell: "default".to_owned(),
+        };
+        let mut hermit = hermit(&config);
+        let mut file_ops = FileOperations::rooted_at(&test_root);
+        let test_files = vec!["bcd", "abc", "cde", "default"];   
+        for x in &test_files {
+            file_ops.create_dir_all(PathBuf::from("shells").join(x));
+        }
+        //assert_eq!(hermit.config.shell_root_path().to_str().unwrap(), "adfdsa");
+        file_ops.commit();
```

Comment:
This is the main reason this test is so much longer.  Ideally, we'll move this work of traversing the `shells` directory to the `FsConfig`. Then that can have an intensive setup that creates several directories and validates we're doing the traversal correctly.  Then in here, we can just depend on the behavior of the `FsConfig` being correct.  In fact, at that point the `Hermit#get_shell_list` method won't need testing because it will be a simple pass through wrapper for the `FsConfig` method.